syntax = "proto3";

package healthcheck;

message HealthCheckRequest {
    string service_name = 1;
}

message HealthCheckResponse {
    enum ServingStatus {
        UNKNOWN = 0;
        SERVING = 1;
        NOT_SERVING = 2;
        SERVICE_UNKNOWN = 3;
    }
    ServingStatus service_status = 1;
}

service HealthCheckService {

    rpc CheckServiceStatus (HealthCheckRequest) returns (HealthCheckResponse);
    rpc WatchServiceStatus (HealthCheckRequest) returns (stream HealthCheckResponse);
}